### ISSUE TITLE
Epic: Create external workspace dashboards or clients (e.g. CLI or IdP) via Public API

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#10427]
Ability to list workspaces for the purposes of creating dashboards or external Gitpod clients, such as CLIs, dashboards, IdP integrations, custom browser plugins, etc. 

**In scope:** 
- Workspace status, 
- HTTP based APIs

**Out of scope**
- Creation of workspaces via API (URL is workaround)

**Relates to:** 
- https://github.com/gitpod-io/gitpod/issues/10798
- https://github.com/gitpod-io/gitpod/issues/15213
- https://github.com/gitpod-io/gitpod/issues/11335
- https://github.com/gitpod-io/gitpod/issues/10799
- https://github.com/gitpod-io/gitpod/issues/14280
- https://github.com/gitpod-io/gitpod/issues/10427
- https://github.com/gitpod-io/gitpod/issues/15419
- https://github.com/gitpod-io/gitpod/issues/15632
- https://github.com/gitpod-io/gitpod/issues/15829

**Callout:** 👋 Are you interested in this feature? We would be interested to [talk to you](https://calendly.com/lou-gitpod/product-feedback). 

**Public API in private beta**:  See [this issue](https://github.com/gitpod-io/gitpod/issues/15453) for more details.

**Comment by jmls** [comment]
is there anything done on this that we could use ? We desperately need a list of workspaces - would be more than happy to be a pre-alpha tester !

**Comment by loujaybee** [comment]
@jmls if you're still interested, please reach out in the Discord community, we have a private beta channel where you can try out some of the APIs in a very early state 🙏

**Comment by loujaybee** [comment]
Moving to future backlog, the private beta remains open, but we're waiting on more customer feedback and input before we progress the various Public API initiatives.
